### ISSUE TITLE
Passing an array to `format` prop doesn't work

### ISSUE THREAD

**Issue by daton89** [gpbl/react-day-picker#763]
i'm trying to format the date value of a form that come from the server, through initialValues of reduxForm options, but the format of the DayPickerInput does not change as expected.

I'm using moment on the client so i'm started from this [example](http://react-day-picker.js.org/api/DayPickerInput#inputProps).

I've setup the format in Italian locale like so:

```jsx
<DayPickerInput
    format="DD/MM/YYYY"
    formatDate={formatDate}
    parseDate={parseDate}
    dayPickerProps={{
        locale: 'it',
        localeUtils: MomentLocaleUtils
    }}
    inputProps={{ ...input }}
    value={input.value}
    onDayChange={input.onChange}
/>
```

So when i initialize the input value with for example `2018-07-27T00:00:00.000Z` i expect the it format the input like this: `27/07/2018`

Instead it come out in the input like the default format: `2018-7-27`

Is this a my misunderstanding or a react-day-picker issue?

I would like to keep the value as a date to save it in the database, and the string formatted value for the input and the daypicker, but i don't know why the value as date is keep only when i change the date, then on focus out of the input the value change in string format... so when try to save the input value sometimes is date other times is string, how can i manage that value? 

i wrote a codesanbox example [here](https://codesandbox.io/s/947vjxp3y)
i wrote a question on stackoverflow [here](https://stackoverflow.com/questions/51566821/how-can-i-format-a-initialized-daypickerinput-of-react-day-picker-with-redux-for)

**Comment by gpbl** [comment]
I've looked better into your example, @daphnesmit . There are some problems in your code:

1. When you write `onDayChange={input.onChange}` you have to remember that the event handler [get a Date object](http://react-day-picker.js.org/api/DayPickerInput#onDayChange) as first argument. So you can't use the `input.onChange` from redux-form in that way. Instead, you have to write

```js
onDayChange={day =>
   input.onChange(day ? formatDate(day, "DD/MM/YYYY") : input.value)
}
```

2. I have no idea what `inputProps={{ ...input }}` is doing from redux-form, but it is giving problems. Better remove it.

Basically, your problem is that you you are using redux-form. I don't understand why it appeals so much developers, it only complicates things :)

Here the updated codebox: https://codesandbox.io/s/kk2kq23855
